File: ruby-bundler/hello-binary/bin/hello.rb
#!/usr/bin/env ruby
# Minimal Ruby executable for the M56 ``ruby-bundler/hello-binary``
# fixture. Bundler's launcher shim (emitted under
# ``<root>/.repro/build/hello/hello.cmd``) invokes
# ``bundle exec ruby bin/hello.rb`` so this file is the actual entry
# point.
#
# Uses only Ruby stdlib (``puts``) so the build needs no external
# gems — keeping the M56 offline-mode contract trivially satisfied.

puts "hello from ruby-bundler-hello-binary"

File: php-composer/hello-binary/bin/hello.php
<?php
# Minimal PHP executable for the M57 ``php-composer/hello-binary``
# fixture. Composer's launcher shim (emitted under
# ``<root>/.repro/build/hello/hello.cmd``) invokes
# ``php bin/hello.php`` so this file is the actual entry point.
#
# Uses only PHP's built-in ``echo`` so the build needs no Packagist
# packages — keeping the M57 offline-mode contract trivially
# satisfied.

echo "hello from php-composer-hello-binary\n";
